filebeat/input/journald/pkg/journalfield/matcher.go

```diff
@@ -118,3 +136,86 @@ func ApplyMatchersOr(j journal, matchers []Matcher) error {
 
 	return nil
 }
+
+// ApplySyslogIdentifierMatcher adds unit based filtering to the journal reader.
+func ApplyUnitMatchers(j journal, units []string, kernel bool) error {
+	for _, unit := range units {
+		matchers := [][]Matcher{
+			[]Matcher{
+				MustBuildMatcher("systemd.unit=" + unit),
+			},
+			[]Matcher{
+				MustBuildMatcher("message_id=fc2e22bc6ee647b6b90729ab34a250b1"),
+				MustBuildMatcher("journald.uid=1"),
+				MustBuildMatcher("journald.coredump.unit=" + unit),
+			},
+			[]Matcher{
+				MustBuildMatcher("journald.pid=1"),
+				MustBuildMatcher("journald.unit=" + unit),
+			},
+			[]Matcher{
+				MustBuildMatcher("journald.uid=1"),
+				MustBuildMatcher("journald.object.systemd.unit=" + unit),
+			},
+		}
+
+		for _, m := range matchers {
+			if err := ApplyMatchersOr(j, m); err != nil {
+				return fmt.Errorf("error while setting up unit matcher for %s: %+v", unit, err)
+			}
+		}
+
+	}
+
+	if kernel {
+		if err := ApplyMatchersOr(j, []Matcher{kernelMatcher}); err != nil {
+			return fmt.Errorf("error while adding kernel transport to matchers: %+v", err)
+		}
+	}
+
+	return nil
+
+}
+
+// ApplySyslogIdentifierMatcher adds syslog identifier filtering to the journal reader.
+func ApplySyslogIdentifierMatcher(j journal, identifiers []string) error {
+	identifierMatchers := make([]Matcher, len(identifiers))
+	for i, identifier := range identifiers {
+		identifierMatchers[i] = MustBuildMatcher("syslog.identifier=" + identifier)
+	}
+
+	return ApplyMatchersOr(j, identifierMatchers)
+}
+
+// ApplyIncludeMatches adds advanced filtering to journals.
+func ApplyIncludeMatches(j journal, m IncludeMatches) error {
+	if len(m.OR) > 0 {
+		for _, or := range m.OR {
+			if err := ApplyIncludeMatches(j, or); err != nil {
+				return err
+			}
+			if err := j.AddDisjunction(); err != nil {
+				return fmt.Errorf("error adding disjunction to journal: %v", err)
+			}
+		}
+	}
+
+	if len(m.AND) > 0 {
+		for _, and := range m.AND {
+			if err := ApplyIncludeMatches(j, and); err != nil {
+				return err
+			}
+			if err := j.AddConjunction(); err != nil {
+				return fmt.Errorf("error adding conjunction to journal: %v", err)
```

Comment:
```suggestion
				return fmt.Errorf("error adding conjunction to journal: %w", err)
```

How about wrapping the errors?